### ISSUE TITLE
User terminology RUM instead of frontend, client-side everywhere.

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#1136]
* Update config to use `rum` over `frontend`, but keep `frontend` backwards compatible.
* Update `v1/client-side` to `v1/rum` officially but also support experimental `v1/client-side` endpoint to be backwards compatible.

**Comment by watson** [comment]
Do we want `v1/rum` to only receive requests from browsers, or will that endpoint also work for native apps for instance?

**Comment by simitt** [comment]
@watson the endpoint could theoretically serve more than browser requests, but needs to be investigated in detail for concrete use cases, as some additional information might make sense for other apps and some current data from the RUM agent might not.

**Comment by watson** [comment]
I was specifically wondering if source maps logic was handled by this endpoint, and if that therefore made this endpoint too browser specific. In that case it might make sense to call it something related to browsers instead. But you know that better.

All I wanted was to avoid that we had to change the name again later when another RUM agent was added and we realized that this endpoint only made sense for browsers.

But I agree that we shouldn't call it something with "client" as you say. We just sometimes have a tendency to simplify it to that RUM = browsers, which obviously is not the case 😄
